libbeat/common/schema/mapstriface/mapstriface.go

```diff
@@ -98,23 +98,23 @@ func (convMap ConvMap) Map(key string, event common.MapStr, data map[string]inte
 		err.Required = convMap.Required
 		return multierror.Errors{err}
 	}
-	subData, ok := d.(map[string]interface{})
-	if !ok {
+	switch subData := d.(type) {
+	case map[string]interface{}:
```

Comment:
we should allow `d` to be `common.MapStr` as well. They are somewhat 'synonymous' in the libbeat at times.